packages/@tailwindcss-vite/src/index.ts

```diff
@@ -261,9 +261,13 @@ function getExtension(id: string) {
 function isPotentialCssRootFile(id: string) {
   let extension = getExtension(id)
   let isCssFile =
-    extension === 'css' ||
-    (extension === 'vue' && id.includes('&lang.css')) ||
-    (extension === 'astro' && id.includes('&lang.css'))
+    (extension === 'css' ||
+      (extension === 'vue' && id.includes('&lang.css')) ||
+      (extension === 'astro' && id.includes('&lang.css'))) &&
+    // Don't intercept ?url and ?raw static asset handling
+    !id.includes('?url') &&
+    !id.includes('?raw')
```

Comment:
Is there ever a case where these are not at the end of the string? If not maybe `endsWith` is clearer?

Reply:
or could combine into a single check with some regex like `.test(/\?(url|raw)$/)`

Reply:
This seems to be the regex Vite uses:

https://github.com/vitejs/vite/blob/257883fccc36f27cc879af249eeba353ebe94cdd/packages/vite/src/node/constants.ts#L86

(Sorry, stumbled across this while searching for something unrelated)

Reply:
@adamwathan yeah I think it can be anywhere in the query string. Doesn't even have to be first place so we also need to look for `&url` or `&raw`. Going with a modified version of the regex instead ty @aaronadamsCA (we don't have to worry about workers/shared workers though)